lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -492,3 +568,84 @@ def disable(self) -> OpenSearchPluginConfig:
                 "s3.client.default.secret_key",
             ],
         )
+
+
+class OpenSearchGCSBackupPlugin(OpenSearchPlugin):
+    """Manage backup configurations for GCS.
+
+    This class must load the opensearch plugin: repository-gcs and configure it.
+    """
+
+    @property
+    def name(self) -> str:
+        """Returns the name of the plugin."""
+        return "repository-gcs"
+
+    def config(self) -> OpenSearchPluginConfig:
+        """Returns OpenSearchPluginConfig composed of configs used at plugin addition.
+
+        Format:
+        OpenSearchPluginConfig(
+            config_entries_to_add = {...},
+            config_entries_to_del = [...]|{...},
+            secret_entries_to_add = {...},
+            secret_entries_to_del = [...]|{...},
+        )
+        """
+        if not self._extra_config.get("service-account") and not self._extra_config.get(
+            "secret-key"
+        ):
+            raise OpenSearchPluginMissingConfigError(
+                "Missing GCS service-account or secret-key configuration",
+                affected_configs=["service-account", "secret-key"],
+                only_log=True,
+            )
+        import base64
+
+        data, conf = (
+            (self._extra_config["service-account"], "service-account")
+            if self._extra_config.get("service-account")
```

Comment:
why do we need the `service-account` if we can do simply with `secret-key` as indicated in the rest of the condition? I don't understand exactly what's happening here 
My understanding was that for the `GCS` plugin, we should exclusively use `service-account` ?

Reply:
yes, on the long run, we should use service-account. However, I do not have support for service-account yet in s3-integrator, so I am passing service account via secret-key for testing.